### PR TITLE
gpu-compute, util-m5: add GPU kernel exit events

### PR DESCRIPTION
**Pull request by yuxiaojia**
The GPUFS scripts include support for dumping and resetting
stats at kernel boundaries by identifying specific GPU kernel 
exit events. This commit extends that support to work with 
GPU SE-mode support.

Change-Id: I662233ae71e2987d90af1fd0100e29036b2ef1c6

### CODE REVIEW

**Comment by mattsinc** [comment]
@yuxiaojia : this should go onto develop, not stable.  Because you tried to add to stable, it grabbed ~400 commits other than yours.

**Comment by mattsinc** [comment]
Also I think we need to use "util-m5" in addition to "gpu-compute" in the commit message.

**Comment by yuxiaojia** [comment]
> Also I think we need to use "util-m5" in addition to "gpu-compute" in the commit message.



> Also I think we need to use "util-m5" in addition to "gpu-compute" in the commit message.

Thanks for pointing that out, I will create a new pull request after my previous PR got merged

**Comment by mattsinc** [comment]
@yuxiaojia : you don't need to wait for #1213 to be merged, you can just redo what you did here but with develop as the target instead

**Comment by mattsinc** [comment]
@yuxiaojia : getting closer.  However it looks like this commit now includes ~8 commits, including other ones part of #1213.  I recommend putting these commits on a separate branch from #1213 to avoid this problem.

**Comment by mattsinc** [comment]
@yuxiaojia : also please update the commit message to be:

gpu-compute: util-m5: add GPU kernel exit events

The GPUFS scripts include support for dump'ing and reset'ing stats at kernel 
boundaries by identifying specific GPU kernel exit events.  This commit
extends that support to work with GPU SE-mode support.

(you may need to update the above to fit in the character limit per line)

**Comment by yuxiaojia** [comment]
> @yuxiaojia : getting closer. However it looks like this commit now includes ~8 commits, including other ones part of #1213. I recommend putting these commits on a separate branch from #1213 to avoid this problem.

I think I made a mistake that I did not create a separate branch for #1213, so I might need to wait for 1213 to get approved then rebase this commit to resolve this problem

**Comment by mattsinc** [comment]
> > @yuxiaojia : getting closer. However it looks like this commit now includes ~8 commits, including other ones part of #1213. I recommend putting these commits on a separate branch from #1213 to avoid this problem.
> 
> I think I made a mistake that I did not create a separate branch for #1213, so I might need to wait for 1213 to get approved then rebase this commit to resolve this problem

I believe if you create a new branch in your repo, remove the commits from #1213, and then force push the branch it will create a new branch without those commits and update this gem5 pull request.  But if this is too complex I can help you with it tomorrow.

**Review by mattsinc — DISMISSED** [review]
@abmerop please check too

File: configs/example/apu_se.py
Comment (abmerop) [inline configs/example/apu_se.py]: You should never see this event since the kernel to skip to is not set in the GPUCommandProcessor above.  I would probably also not dump stats here and just continue as no kernel was simulated, however I don't know the exact use case you are going for.

**Review by abmerop — DISMISSED** [review]

File: configs/example/apu_se.py
Comment (mattsinc) [reply configs/example/apu_se.py]: Yeah, @yuxiaojia this should probably be removed then -- @abmerop did the kernel skip only get added for GPUFS?

Comment (abmerop) [reply configs/example/apu_se.py]: Yes, it looks like it was only added for GPUFS

File: .github/workflows/daily-tests.yaml
Comment (mattsinc) [inline .github/workflows/daily-tests.yaml]: @yuxiaojia are these changes required as part of your update?

File: src/base/loader/symtab.test.cc
Comment (mattsinc) [inline src/base/loader/symtab.test.cc]: Same thing -- this doesn't seem part of your change

File: src/mem/ruby/protocol/GPU_VIPER-TCC.sm
Comment (mattsinc) [inline src/mem/ruby/protocol/GPU_VIPER-TCC.sm]: This is part of a separate commit

**Review by mattsinc — CHANGES_REQUESTED** [review]
It looks like you merged in other changes when updating -- please fix.

File: configs/example/apu_se.py
Comment (mattsinc) [inline configs/example/apu_se.py]: Looks like this still needs to be removed

**Review by mattsinc — CHANGES_REQUESTED** [review]

**Review by mattsinc — APPROVED** [review]

**Review by abmerop — APPROVED** [review]

**Pull request merged**
